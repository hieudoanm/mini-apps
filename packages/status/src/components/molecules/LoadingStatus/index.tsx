import Status from '../../atoms/Status';

type LoadingStatusProps = {
  loading: boolean;
  error: any;
  data: any;
  active: boolean;
  service: string;
  refetch: () => void;
};

const LoadingStatus: React.FC<LoadingStatusProps> = ({
  loading,
  error,
  data,
  active,
  service,
  refetch,
}) => {
  if (loading) {
    return (
      <Status
        bgColor={'bg-gray-500'}
        status={'Loading'}
        service={service}
        refetch={refetch}
      />
    );
  }

  if (error) {
    return (
      <Status
        bgColor={'bg-red-500'}
        status={'Error'}
        service={service}
        refetch={refetch}
      />
    );
  }

  if (!data) {
    return (
      <Status
        bgColor={'bg-red-500'}
        status={'No Data'}
        service={service}
        refetch={refetch}
      />
    );
  }

  const status: string = active ? 'Active' : 'Inactive';
  const bgColor: string = active ? 'bg-green-500' : 'bg-red-500';

  return (
    <Status
      bgColor={bgColor}
      status={status}
      service={service}
      refetch={refetch}
    />
  );
};

LoadingStatus.displayName = 'LoadingStatus';

export default LoadingStatus;
