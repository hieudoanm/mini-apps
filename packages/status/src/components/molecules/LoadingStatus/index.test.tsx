import { render } from '@testing-library/react';
import LoadingStatus from '.';

describe('LoadingStatus', () => {
  test('render loading', () => {
    const wrapper = render(
      <LoadingStatus
        service="service"
        refetch={() => {}}
        loading={true}
        error={null}
        data={{}}
        active={false}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  test('render error', () => {
    const wrapper = render(
      <LoadingStatus
        service="service"
        refetch={() => {}}
        loading={false}
        error={{ message: 'Error' }}
        data={{}}
        active={false}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  test('render null data', () => {
    const wrapper = render(
      <LoadingStatus
        service="service"
        refetch={() => {}}
        loading={false}
        error={null}
        data={null}
        active={false}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  test('render active', () => {
    const wrapper = render(
      <LoadingStatus
        service="service"
        refetch={() => {}}
        loading={false}
        error={null}
        data={{ status: 'ok' }}
        active={true}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  test('render inactive', () => {
    const wrapper = render(
      <LoadingStatus
        service="service"
        refetch={() => {}}
        loading={false}
        error={null}
        data={{ status: 'ok' }}
        active={false}
      />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
