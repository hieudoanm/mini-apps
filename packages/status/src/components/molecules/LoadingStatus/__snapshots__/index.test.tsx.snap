// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingStatus render active 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      service
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            Active
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-green-500"
          />
        </div>
      </div>
      <button
        class="rounded bg-red-500 text-white px-2 py-1 text-sm"
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
</div>
`;

exports[`LoadingStatus render error 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      service
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            Error
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-red-500"
          />
        </div>
      </div>
      <button
        class="rounded bg-red-500 text-white px-2 py-1 text-sm"
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
</div>
`;

exports[`LoadingStatus render inactive 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      service
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            Inactive
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-red-500"
          />
        </div>
      </div>
      <button
        class="rounded bg-red-500 text-white px-2 py-1 text-sm"
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
</div>
`;

exports[`LoadingStatus render loading 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      service
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            Loading
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-gray-500"
          />
        </div>
      </div>
      <button
        class="rounded bg-red-500 text-white px-2 py-1 text-sm"
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
</div>
`;

exports[`LoadingStatus render null data 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      service
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            No Data
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-red-500"
          />
        </div>
      </div>
      <button
        class="rounded bg-red-500 text-white px-2 py-1 text-sm"
        type="button"
      >
        Refresh
      </button>
    </div>
  </div>
</div>
`;
